pkg/scalers/nsq_scaler.go

```diff
@@ -0,0 +1,358 @@
+	for nsqdHost := range nsqdHostMap {
+		nsqdHosts = append(nsqdHosts, nsqdHost)
+	}
+
+	return nsqdHosts, nil
+}
+
+func (s *nsqScaler) getLookup(host string, topic string) (*lookupResponse, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s/%s", host, "lookup"), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Accept", "application/json; charset=utf-8")
+
+	params := url.Values{"topic": {topic}}
+	req.URL.RawQuery = params.Encode()
+
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code '%s'", resp.Status)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var lookupResponse lookupResponse
+	err = json.Unmarshal(body, &lookupResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return &lookupResponse, nil
+}
+
+type statsResponse struct {
+	Topics []struct {
+		TopicName string `json:"topic_name"`
+		Depth     int64  `json:"depth"`
+		Channels  []struct {
+			ChannelName string `json:"channel_name"`
+			Depth       int64  `json:"depth"`  // num messages in the queue (mem + disk)
+			Paused      bool   `json:"paused"` // if paused, consumers will not receive messages
+		}
+	}
+}
+
+type statsResult struct {
+	host          string
+	statsResponse *statsResponse
+	err           error
+}
+
+func (s *nsqScaler) aggregateDepth(nsqdHosts []string, topic string, channel string) (int64, error) {
+	wg := sync.WaitGroup{}
+	resultCh := make(chan statsResult, len(nsqdHosts))
+
+	for _, host := range nsqdHosts {
+		wg.Add(1)
+		go func(host string, topic string) {
+			defer wg.Done()
+			resp, err := s.getStats(host, topic)
+			resultCh <- statsResult{host, resp, err}
+		}(host, topic)
+	}
+
+	wg.Wait()
+	close(resultCh)
+
+	var depth int64
+	for result := range resultCh {
+		if result.err != nil {
+			return -1, fmt.Errorf("error getting stats from host '%s': %w", result.host, result.err)
+		}
+
+		for _, t := range result.statsResponse.Topics {
+			if t.TopicName != topic {
+				// this should never happen as we make the /stats call with the "topic" param
+				continue
+			}
+
+			if len(t.Channels) == 0 {
+				// topic exists with no channels, but there are messages in the topic -> we should still scale to bootstrap
+				s.logger.Info("no channels exist for topic", "topic", topic, "channel", channel, "host", result.host)
+				depth += t.Depth
+				continue
+			}
+
+			channelExists := false
+			for _, ch := range t.Channels {
+				if ch.ChannelName != channel {
+					continue
+				}
+				channelExists = true
+				if ch.Paused {
+					// if it's paused on a single nsqd host, it's depth should not go into the aggregate
+					// meaning if paused on all nsqd hosts => depth == 0
+					s.logger.Info("channel is paused", "topic", topic, "channel", channel, "host", result.host)
```

Comment:
This should be a debug info:
```suggestion
					s.logger.V(1).Info("channel is paused", "topic", topic, "channel", channel, "host", result.host)
```